### PR TITLE
[ALLUXIO-3306] delay fuse open if file is not completed

### PR DESCRIPTION
**Pull request by LuQQiu**
https://alluxio.atlassian.net/browse/ALLUXIO-3306

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java [hunk 1]
```diff
@@ -63,6 +66,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(AlluxioFuseFileSystem.class);
 
   private static final int MAX_OPEN_FILES = Integer.MAX_VALUE;
+  private static final int MAX_OPEN_WAITTIME = 5000;
```
Comment (calvinjia) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java]: Could you suffix this with `MS` since the units are unclear?

**Comment by LuQQiu** [comment]
@calvinjia Changed the constant name, PTAL, thanks!

**Comment by LuQQiu** [comment]
@calvinjia Added the unit test for delayed open, change file completed status when opening a file. PTAL, thanks

File: integration/fuse/src/test/java/alluxio/fuse/AlluxioFuseFileSystemTest.java [hunk 2]
```diff
@@ -174,9 +175,18 @@ public void open() throws Exception {
...
+    Thread t = new Thread(() -> mFuseFs.open("/foo/bar", mFileInfo));
+    t.start();
+    Thread.sleep(1000);
+    // If the file exists but is not completed, we will wait for the file to complete
     verify(mFileSystem).exists(expectedPath);
+    verify(mFileSystem, atLeast(10)).getStatus(expectedPath);
+    verify(mFileSystem, never()).openFile(expectedPath);
+
+    fi.setCompleted(true);
+    t.join();
```
Comment (calvinjia) [inline integration/fuse/src/test/java/alluxio/fuse/AlluxioFuseFileSystemTest.java]: Could you add this test separate from the existing open test, ideally we test one case in each unit test.

**Pull request merged**